flink-kubernetes/src/test/java/org/apache/flink/kubernetes/KubernetesResourceManagerDriverTest.java

```diff
@@ -223,6 +225,27 @@ void testFatalHandleError() throws Exception {
         };
     }
 
+    @Test
+    void testKubernetesExceptionHandling() throws Exception {
+        new Context() {
+            {
+                runTest(
+                        () -> {
+                            runInMainThread(
+                                            () -> {
+                                                getDriver()
+                                                        .requestResource(TASK_EXECUTOR_PROCESS_SPEC)
+                                                        .completeExceptionally(
+                                                                new CompletionException(
+                                                                        new KubernetesClientException(
+                                                                                "test")));
+                                            })
+                                    .get(TIMEOUT_SEC, TimeUnit.SECONDS);
```

Comment:
nit: It's not very intuitive to assert that the future has not completed exceptionally via `.get(TIMEOUT_SEC, TimeUnit.SECONDS)`, how about directly using `future assertion`.
```
runTest(
                        () ->
                                FlinkAssertions.assertThatFuture(
                                                runInMainThread(
                                                        () -> {
                                                            getDriver()
                                                                    .requestResource(
                                                                            TASK_EXECUTOR_PROCESS_SPEC)
                                                                    .completeExceptionally(
                                                                            new CompletionException(
                                                                                    new KubernetesClientException(
                                                                                            "test")));
                                                        }))
                                        .eventuallySucceeds());
```

Reply:
Yea that looks better, I will clean that up 👍